de.fhdo.lemma.servicedsl.extractor/src/de/fhdo/lemma/servicedsl/extractor/ServiceDslExtractor.xtend

```diff
@@ -0,0 +1,272 @@
+package de.fhdo.lemma.servicedsl.extractor
+
+import de.fhdo.lemma.data.ComplexType
+import de.fhdo.lemma.data.PrimitiveType
+import de.fhdo.lemma.service.Endpoint
+import de.fhdo.lemma.service.Import
+import de.fhdo.lemma.service.ImportType
+import de.fhdo.lemma.service.ImportedType
+import de.fhdo.lemma.service.Interface
+import de.fhdo.lemma.service.Microservice
+import de.fhdo.lemma.service.MicroserviceType
+import de.fhdo.lemma.service.Operation
+import de.fhdo.lemma.service.Parameter
+import de.fhdo.lemma.service.ServiceModel
+import de.fhdo.lemma.service.Visibility
+import de.fhdo.lemma.technology.CommunicationType
+import de.fhdo.lemma.technology.ExchangePattern
+import de.fhdo.lemma.technology.Protocol
+import de.fhdo.lemma.service.ImportedServiceAspect
+
+
+/**
+ * Model-to-text extractor for the Service DSL.
+ *
+ * @author <a href="mailto:jonas.sorgalla@fh-dortmund.de">Jonas Sorgalla</a>
+ */
+class ServiceDslExtractor {
+    static val IMPORTED_TECH_ALIASES = <String>newArrayList()
+    static val ID_PATTERN = "(\\^?)([a-zA-Z_])\\w*"
+    static val QUALIFIED_NAME_PATTERN = '''«ID_PATTERN»(\.«ID_PATTERN»)*'''
+    static val QUALIFIED_NAME_WITH_AT_LEAST_ONE_LEVEL_PATTERN = '''«ID_PATTERN»\.''' +
+        QUALIFIED_NAME_PATTERN
+
+    /**
+     * Extract ServiceModel
+     */
+    def String extractToString(ServiceModel serviceModel) {
+        val imports = <String>newArrayList
+        serviceModel.imports.forEach[imports.add(it.generate)]
+        val importStatements = if (!imports.empty)
+                String.join("\n", imports) + "\n\n"
+            else
+                ""
+
+        val microservices = String.join("\n\n", serviceModel.microservices.map[generate])
+
+        return '''«importStatements»«microservices»'''
+    }
+
+    /**
+     * Extract Imports
+     */
+    private def generate(Import ^import) {
+        val importTypeKeyword = switch(^import.importType) {
+            case DATATYPES: "datatypes"
+            case MICROSERVICES: "microservices"
+            case TECHNOLOGY: "technology"
+            default: null
```

Comment:
Throw an `IllegalArgumentException` here to fail early and not produce wrong LEMMA model code when there is an unsupported (maybe then-new) import type.